jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/jpa/PersistenceProvider.java

```diff
@@ -316,29 +320,59 @@ public JPAInitializer getInitializer(String emName, Map m){
         return JavaSECMPInitializer.getJavaSECMPInitializer(classLoader);
     }
 
+    // Package visibility is required for jUnit
     /**
-     * Need to check that the provider property is null or set for EclipseLink
+     * The Persistence bootstrap class must locate all the persistence providers using the PersistenceProviderResolver
+     * mechanism described in Section 9.3 and call createEntityManagerFactory on them in turn until an appropriate backing
+     * provider returns an EntityManagerFactory instance. A provider may deem itself as appropriate for the persistence unit
+     * if any of the following are true:
+     * <p> * Its implementation class has been specified in the provider element for that persistence unit in the persistence.xml
+     * file and has not been overridden by a different jakarta.persistence.provider property value included in the Map passed
+     * to the createEntityManagerFactory method.
+     * <p> * The jakarta.persistence.provider property was included in the Map passed to createEntityManagerFactory and the value
+     * of the property is the provider’s implementation class.
+     * <p> * No provider was specified for the persistence unit in either the persistence.xml or the property map.
+     *
+     * @since Jakarta Persistence 3.2
      */
-    public boolean checkForProviderProperty(Map properties){
-        Object provider = properties.get("jakarta.persistence.provider");
```

Comment:
Hard coded String `"jakarta.persistence.provider"` instead of reference `PersistenceUnitProperties.PROVIDER`.

Reply:
Exactly. Red marked code is the one being removed. And there was no `PersistenceUnitProperties.PROVIDER` before this changeset.
That's why I  added it into `PersistenceUnitProperties` and changed hardcoded valut to the static constant reference. See the green part of the diff.